plugin/externalPluginEntry.go

```diff
@@ -96,34 +147,55 @@ func (e *externalPluginEntry) Schema() *EntrySchema {
 	return nil
 }
 
+func (e *externalPluginEntry) HasSourceParent() bool {
+	return e.hasSourceParent
+}
+
+const listFormat = "[{\"name\":\"entry1\",\"methods\":[\"list\"]},{\"name\":\"entry2\",\"methods\":[\"list\"]}]"
+
 func (e *externalPluginEntry) List(ctx context.Context) ([]Entry, error) {
-	stdout, err := e.script.InvokeAndWait(ctx, "list", e)
-	if err != nil {
-		return nil, err
-	}
 	var decodedEntries []decodedExternalPluginEntry
-	if err := json.Unmarshal(stdout, &decodedEntries); err != nil {
-		return nil, newStdoutDecodeErr(
-			ctx,
-			"the entries",
-			err,
-			stdout,
-			"[{\"name\":\"entry1\",\"methods\":[\"list\"]},{\"name\":\"entry2\",\"methods\":[\"list\"]}]",
-		)
+	if impl, ok := e.methods["list"]; ok && impl != nil {
+		// Entry statically implements list. Construct new entries based on that rather than invoking the script.
+		bits, err := json.Marshal(impl)
```

Comment:
Guessing early validation of list's result didn't work out?

Reply:
That'd be a lot of extra validation. I didn't explore it.

Reply:
That's fine, was just curious.

Reply:
Not really. You'd want to try it out by actually listing things, so you'd run into an error pretty quickly.